src/lib.rs

```diff
@@ -266,8 +268,17 @@ impl<'a> Executor<'a> {
     fn schedule(&self) -> impl Fn(Runnable) + Send + Sync + 'static {
         let state = self.state().clone();
 
-        // TODO: If possible, push into the current local queue and notify the ticker.
-        move |runnable| {
+        move |mut runnable| {
+            // If possible, push into the current local queue and notify the ticker.
+            if let Some(local) = state.local_queue.get() {
+                runnable = if let Err(err) = local.queue.push(runnable) {
+                    err.into_inner()
+                } else {
+                    local.waker.wake();
+                    return;
```

Comment:
An issue I've realized here; this only wakes up the local thread runner. Ideally we would wake up another thread if the current thread is not sleeping. Consider the following scenario:

- Thread 1 and Thread 2 are calling `run()`.
- Task A is spawned on Thread 1.
- While running Task A, it spawns Task B. Task B is pushed to the local queue; however since the runner is awke calling `local.waker.wake()` does nothing.
- Task A is taking a while and preventing Thread 1 from handling any other tasks.

Ideally spawning Task B would then wake up Thread 2 so it can steal the task from Thread 1.

I think the best way of doing this would be to add a method to `Sleepers` that takes a unique ticket ID, tries to wake up that ticker, but then wakes up another ticket if that one is already awake.

Reply:
`AtomicWaker::take` should return a `Option<Waker>`, which we can just call `notify` if it's `None`, right? Is there a reason to avoid this approach?

Reply:
I forgot about `take`, that's probably a better idea.

Reply:
Done!